src/test/codewhisperer/commands/basicCommands.test.ts

```diff
@@ -0,0 +1,121 @@
+                targetCommand.dispose()
+            }
+            sinon.restore()
+        })
+
+        it('test get()', async function () {
+            const fakeMemeto = new FakeMemento()
+            fakeMemeto.update(CodeWhispererConstants.autoTriggerEnabledKey, true)
+
+            let res = get(CodeWhispererConstants.autoTriggerEnabledKey, fakeMemeto)
+            assert.strictEqual(res, true)
+
+            fakeMemeto.update(CodeWhispererConstants.autoTriggerEnabledKey, undefined)
+            res = get(CodeWhispererConstants.autoTriggerEnabledKey, fakeMemeto)
+            assert.strictEqual(res, undefined)
+
+            fakeMemeto.update(CodeWhispererConstants.autoTriggerEnabledKey, false)
+            res = get(CodeWhispererConstants.autoTriggerEnabledKey, fakeMemeto)
+            assert.strictEqual(res, false)
+        })
+
+        it('test set()', async function () {
+            const fakeMemeto = new FakeMemento()
+            set(CodeWhispererConstants.autoTriggerEnabledKey, true, fakeMemeto)
+            assert.strictEqual(fakeMemeto.get(CodeWhispererConstants.autoTriggerEnabledKey), true)
+
+            set(CodeWhispererConstants.autoTriggerEnabledKey, false, fakeMemeto)
+            assert.strictEqual(fakeMemeto.get(CodeWhispererConstants.autoTriggerEnabledKey), false)
+        })
+
+        it('test toggleCodeSuggestions: should emit aws_modifySetting event on user toggling autoSuggestion -- activate', async function () {
+            const fakeExtContext: ExtContext = {
+                extensionContext: await FakeExtensionContext.create(),
+                awsContext: {} as AwsContext,
+                samCliContext: () => {
+                    return {} as SamCliContext
+                },
+                regionProvider: {} as RegionProvider,
+                outputChannel: {} as vscode.OutputChannel,
+                telemetryService: {} as TelemetryService,
+                credentialsStore: {} as CredentialsStore,
+                invokeOutputChannel: {} as vscode.OutputChannel,
+            }
+            targetCommand = testCommand(toggleCodeSuggestions, fakeExtContext)
+
+            assert.strictEqual(
+                fakeExtContext.extensionContext.globalState.get(CodeWhispererConstants.autoTriggerEnabledKey),
+                undefined
+            )
+            await targetCommand.execute()
+            const res = fakeExtContext.extensionContext.globalState.get(CodeWhispererConstants.autoTriggerEnabledKey)
+            assert.strictEqual(res, true)
+            assertTelemetryCurried('aws_modifySetting')({
+                settingId: CodeWhispererConstants.AutoSuggestion.settingId,
+                settingState: CodeWhispererConstants.AutoSuggestion.activated,
+            })
+        })
+
+        it('test toggleCodeSuggestions: should emit aws_modifySetting event on user toggling autoSuggestion - deactivate', async function () {
+            const fakeExtContext: ExtContext = {
```

Comment:
Once we change the command input to `vscode.Memento` we can use `FakeMemento` instead.

Reply:
yea